src/lustre/lustreAstDependencies.ml

```diff
@@ -478,7 +481,7 @@ let rec extract_decls: ('a IMap.t * id_pos_map) -> LA.ident list -> ('a list) gr
   | i :: is ->
      (match (IMap.find_opt i decl_map) with
       | None -> (match (find_id_pos i_pos_map i) with
-                 | None -> failwith ("Identifier " ^ i ^ " not found. This should not happen")
+                 | None -> fail_no_position ("Identifier " ^ i ^ " not found. This should not happen")
```

Comment:
Why did you replace `failwith` with `fail_no_position` which has the side-effect of printing a message? One of the goals of the new design is to make the code free of side effects so that messages are only printed at the top-level.

Reply:
All `failwith` exceptions should be caught within the module and be notified through the error part of the `graph_result`

Reply:
Getting a None here is a panic state and we do not expect it to happen.
currently `failwith` raises an obscure error which gets caught in `Kind2.ml` generic exception handler.
I had changed it to `fail_no_pos` so that it fails fast but prints a more pretty error.

I have changed it to `graph_error` now.